### PR TITLE
Handle canceling of jobs

### LINKED ISSUE #127: [Integration] Handle Job Cancelation
AC: When user clicks the `Cancel` button corresponding to a running job, the job should be updated with status `Canceled` and be moved to the `Completed Jobs` table.

### PR DESCRIPTION
**Pull request by ShriramG24**
Closes #127.

### CODE REVIEW

**Review by atharvakale343 — CHANGES_REQUESTED** [review]

File: src/main/model-apps/job-manager.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import log from 'electron-log';
+import InferenceService, {
+  InferenceArgs,
+  SuccessResponse,
+  ErrorResponse,
+  ModelServerInfo,
+} from './inference-service';
+
+class JobManager {
```
[the remark this answers is not in the document]
Comment (ShriramG24) [reply src/main/model-apps/job-manager.ts]: Sure, makes sense.

File: src/renderer/Jobs.tsx [hunk 2]
```diff
@@ -113,10 +113,7 @@ function Jobs() {
             </TableHeader>
             <TableBody>
               {jobs
-                ?.filter(
-                  (job) =>
-                    job.status !== 'Completed' && job.status !== 'Failed',
-                )
+                ?.filter((job) => job.status === 'Running')
```
[the remark this answers is not in the document]
Comment (ShriramG24) [reply src/renderer/Jobs.tsx]: Turns out there's no issues even without the `?`.

File: src/renderer/Jobs.tsx [hunk 3]
```diff
@@ -177,7 +171,7 @@ function Jobs() {
                       {job.endTime ? job.endTime.toUTCString() : 'N/A'}
                     </TableCell>
                     <TableCell className="w-1/6 pl-6">
-                      {job.status === 'Failed' ? (
+                      {job.status === 'Failed' || job.status === 'Canceled' ? (
```
[the remark this answers is not in the document]
Comment (ShriramG24) [reply src/renderer/Jobs.tsx:172]: Will do.

**Comment by ShriramG24** [comment]
Updated status icons.

image

**Review by atharvakale343 — APPROVED** [review]
LGTM, nice!

**Pull request merged**